src/rpc/blockchain.cpp

```diff
@@ -1318,29 +1309,25 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
-            "     {\n"
-            "        \"id\": \"xxxx\",           (string) name of softfork\n"
-            "        \"version\": xx,          (numeric) block version\n"
-            "        \"reject\": {             (object) progress toward rejecting pre-softfork blocks\n"
-            "           \"status\": xx,        (boolean) true if threshold reached\n"
-            "        },\n"
-            "     }, ...\n"
-            "  ],\n"
-            "  \"bip9_softforks\": {           (object) status of BIP9 softforks in progress\n"
+            "  \"softforks\": {                (object) status of softforks\n"
             "     \"xxxx\" : {                 (string) name of the softfork\n"
+            "      \"type\": \"xxxx\",           (string) one of \"buried\", \"bip9\"\n"
+            "      \"bip9\": {                 (object) status of bip9 softforks (only for \"bip9\" type)\n"
             "        \"status\": \"xxxx\",       (string) one of \"defined\", \"started\", \"locked_in\", \"active\", \"failed\"\n"
             "        \"bit\": xx,              (numeric) the bit (0-28) in the block version field used to signal this softfork (only for \"started\" status)\n"
             "        \"startTime\": xx,        (numeric) the minimum median time past of a block at which the bit gains its meaning\n"
             "        \"timeout\": xx,          (numeric) the median time past of a block at which the deployment is considered failed if not yet locked in\n"
             "        \"since\": xx,            (numeric) height of the first block to which the status applies\n"
-            "        \"statistics\": {         (object) numeric statistics about BIP9 signalling for a softfork (only for \"started\" status)\n"
+            "        \"statistics\": {         (object) numeric statistics about BIP9 signalling for a softfork\n"
             "           \"period\": xx,        (numeric) the length in blocks of the BIP9 signalling period \n"
             "           \"threshold\": xx,     (numeric) the number of blocks with the version bit set required to activate the feature \n"
             "           \"elapsed\": xx,       (numeric) the number of blocks elapsed since the beginning of the current period \n"
             "           \"count\": xx,         (numeric) the number of blocks with the version bit set in the current period \n"
             "           \"possible\": xx       (boolean) returns false if there are not enough blocks left in this period to pass activation threshold \n"
             "        }\n"
+            "      },\n"
+            "      \"height\": \"xxxxxx\",       (numeric) height of the first block which the rules are enforced (only for \"buried\" type, or \"bip9\" type with \"locked_in\" or \"active\" status)\n"
+            "      \"active\": xx,             (boolean) true if the rules are enforced at the current chain tip\n"
```

Comment:
After sleeping on it, I think I've changed my mind -- it's probably less confusing to have "active=true" tell you whether the rules were enforced on the current block, and you can use the "height=N" output to tell you whether it needs to be enforced on the next block.

In that case, I think you'd want `nHeight >= BIPxxHeight` and to use `tip->pprev` instead of `tip` for the versionbits state. The versionbits statistics function might need some more tweaking to make sense though.

Reply:
I've pushed an update to https://github.com/ajtowns/bitcoin/commits/201907-getforkinfo that implements the latter in a way I think makes sense, including changes to versionist statistics. Can check the output makes sense with something like (assuming the "csv" softfork activates at block 1152 and you have jshon available):

    $ for a in 863 864 865 1006 1007 1008 1009 1010 1150 1151 1152; do echo $a; bitcoin-cli -regtest getforkinfo $(bitcoin-cli -regtest getblockhash $a)  | jshon -e csv; echo ======; done | less